### ISSUE TITLE
What are the plans for this library?

### ISSUE THREAD

**Issue by voslartomas** [mjarkk/go-graphql#2]
Hello, I really like your approach, what are your plans? Is this just hobby project or you are planning to use it in production environment? Also are you considering support for Apollo Federation (probably only thing needed is support of directives).

**Comment by redbar0n** [comment]
FWIW, @mjarkk I recommend collaborating with @sijad which has made ‘gentle’, a library very similar to ‘go-graphql’. If you find that your goals overlap.

https://github.com/sijad/gentle/issues/1#issuecomment-898593298

**Comment by mjarkk** [comment]
Oh hello, glad you like my project :)

> what are your plans?

Currently i'm re-writing the whole backend to make it faster and have no changes for race conditions *(this will probably take a while)*. 
For personal plans see:
https://github.com/mjarkk/go-graphql/projects/2
https://github.com/mjarkk/go-graphql/projects/3
Beside that it's mostly bug fixes, code clarification and user input on other things to add.

> Is this just hobby project or you are planning to use it in production 

This is a hobby project tough i am making this to be used in a production environment.
**Note that the library is currently not production ready as it has race conditions.**

> Also are you considering support for Apollo Federation

That's new to me, if i only need to add support for user defined directives than that's probably supported in the further

**Comment by mjarkk** [comment]
Just released [v0.5.0](https://github.com/mjarkk/go-graphql/releases/tag/v0.5.0), this is the first release with the fully re-written core.
This also adds support for custom directives they are still quite basic but it's a nice start

**Comment by mjarkk** [comment]
Just benchmarked the library compared to others and the result seems very promising:

This is the result of [appleboy/golang-graphql-benchmark](https://github.com/appleboy/golang-graphql-benchmark) with `resolver_benchmark_test.go > BenchmarkHelloWorldResolve` added:

_The last one is this library_

```sh
# go test -v -bench=Master -benchmem
# goos: darwin
# goarch: amd64
# pkg: github.com/appleboy/golang-graphql-benchmark
# cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkGoGraphQLMaster
BenchmarkGoGraphQLMaster-12          	   24992	     48180 ns/op	   26895 B/op	     445 allocs/op
BenchmarkPlaylyfeGraphQLMaster-12    	  320289	      3770 ns/op	    2797 B/op	      57 allocs/op
BenchmarkGophersGraphQLMaster-12     	  391269	      3114 ns/op	    3634 B/op	      38 allocs/op
BenchmarkThunderGraphQLMaster-12     	  708327	      1707 ns/op	    1288 B/op	      30 allocs/op
BenchmarkMjarkkGraphQLGoMaster-12    	 2560764	       466.5 ns/op	      80 B/op	       1 allocs/op
```
